### ISSUE TITLE
Make it easier to use the signing secret returned by --forward-to

### ISSUE THREAD

**Issue by guanzo** [stripe/stripe-cli#472]
### Problem
AFAIK, the only way to get the secret is to copy/paste from the terminal output. Now where do I put this secret? 
* I can put it in my source, which triggers git changes.
* I can put it in my env file, which is okay-ish. It's laborious for me because my env file is pulled from a server, so I have to jump thru several hoops to update my env file.
* I can set it as an env var in my current shell. I can't set it as a global env var, I have other projects that use stripe webhooks.
* I can put it in X...

The problem with all these options is I have to **manually** do something to make the secret accessible to my code. Also, there'll be conflicts with other devs if we all put our own secrets into a shared repo/file. Also, the secret changes every so often, so I have to repeat this admittedly quick task. I'm lazy, I don't want to deal with this.

### Feature
Have the cli set the secret to an environment variable like `STRIPE_WEBHOOK_SECRET`.
Then I can just do this, and it'll work for dev and prod.

```js
stripe.webhooks.constructEvent(req.body, sig, process.env.STRIPE_WEBHOOK_SECRET)
```

### Examples

My workflow is running a shell script that runs `stripe listen` in the background, and also starts my node server. Here's what I'd like to see.

```sh
λ ./start-dev-services.sh
A newer version of the Stripe CLI is available, please update to: v1.4.3
> Ready! Your webhook signing secret is whsec_W6VEw3GMMdIrd77uBfQM25WqRpvBJbjv
> The secret has been set to the environment variable STRIPE_WEBHOOK_SECRET
```

Then my code runs, checks for `process.env.STRIPE_WEBHOOK_SECRET`, and it *just works* for dev and prod, b/c in prod the secret is also in an env variable.

This should be a cli option, like `stripe listen --env-key=WHATEVER_I_WANT`, then `process.env.WHATEVER_I_WANT` contains the secret.

**Comment by richardm-stripe** [comment]
Hello @guanzo, thank you for the suggestion. I like the idea of making it easier to return the signing secret. I'm not sure `--env-key` is a viable though, because child processes (the stripe cli) do not generally have the ability to alter the environment variables of their parent process (in this case, your `start-dev-services.sh`) -- as far as I am aware, anyway. Do you know of any other tools that operate this way?

We could add the ability to write to a file, i.e.
```bash
stripe listen --forward-to localhost:3000 --write-signing-secret-to /tmp/signingsecret
```
and then in your `start-dev-services.sh` you would do something like
```bash
stripe listen --forward-to localhost:3000 --write-signing-secret-to /tmp/signingsecret &
sleep 5
export STRIPE_WEBHOOK_SECRET=$(cat /tmp/signingsecret)
# start your application
```

But I'm not sure this is much of an improvement, considering today you could also do something like

`stripe_wrapper.sh`
```bash
#!/usr/bin/env bash
ss_file="$(mktemp)"
stripe listen --forward-to localhost:3000 2> "${ss_file}" 1> /dev/null &
while ! < "${ss_file}" grep -o "whsec_[^ ]*"; do sleep 1; done
rm "${ss_file}"
```

`start-dev-services.sh`
```bash
#!/usr/bin/env bash
export STRIPE_WEBHOOK_SECRET=$(./stripe_wrapper.sh)
# start your application
```

(although this is missing a process for cleaning up the background `stripe listen` processes).

Curious what your thoughts are -- would writing to a file still be an improvement? And OOC how do you handle cleaning background processes in your workflow?

**Comment by guanzo** [comment]
Thank you for the detailed reply.

> child processes (the stripe cli) do not generally have the ability to alter the environment variables of their parent process

Hmm.. good point. My usual workflow does allow child processes to alter the parent env, because my shell scripts have a lot of this:


load-env-vars.sh  
```sh
#!/usr/bin/env bash

set -a 
tmp=$(mktemp)
git archive --remote=some-git-url-that-returns-an-env-file | tar -xO > $tmp
source $tmp 
rm $tmp
set +a
```

start-dev-services.sh 
```sh
#!/usr/bin/env bash

source load-env-vars.sh  

# env vars from the file are now available in the current env, and do not leak into the terminal's env.
npm run dev
```

> Do you know of any other tools that operate this way?

Nope, I've never seen a cli do this, it was a bad idea, sorry. My "vision of the dev experience" is heavily skewed to my work environment. I'm not sure how this flow would work the the stripe cli. 

> But I'm not sure this is much of an improvement, considering today you could also do something like

See, I knew it was technically possible to scrape the stdout for the key, but I'm not strong with the cmd line and shell scripting. That code is borderline gibberish to me. That's why I was asking for a relatively simple solution. I will try out `stripe_wrapper.sh`, thanks for the example.

> Curious what your thoughts are -- would writing to a file still be an improvement?

Hmm.. it would solve my issue, but it doesn't feel clean to me. 

> And OOC how do you handle cleaning background processes in your workflow?

here's trimmed down version of `start-dev-services.sh`. Again, my shell scripting is amateur at best. I struggle to get something working, then leave it as is.

```sh
#!/usr/bin/env bash

set -e

scriptDir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
projectDir="$scriptDir/../"

source "$scriptDir/../../bin/load-dev-env-vars.sh"

stripe listen --skip-verify --api-key=$STRIPE_SECRET_KEY \
--forward-to https://.localhost:7055/stripe-webhook &

cd "$projectDir"

npm run dev &
wait
```

What I think this does, is run `stripe` and `npm run dev` in the background, then waits... for something. I can then enter `Ctrl + C` in the terminal to kill everything. Sometimes I have to enter it multiple times to actually kill all the child processes lol. Any tips to improve my workflow would be **greatly** appreciated.

Long story short, I don't want to copy/paste the key from the terminal. I want an automated way to make the key available to my node process.

**Comment by bwang-stripe** [comment]
Hi @guanzo,

Thanks again for bringing this up! We've been discussing internally, and we've settled on adding a `--print-secret` flag to `stripe listen` that would directly print out the webhook signing secret. From there, you could use that to set a local environment variable, or shell out from your code to get the secret. 

I've created #475 to track the progress on that feature, so if you have anything to add to the feature request there, please do so! 

I believe having this should resolve the issues with your workflow?

**Comment by bwang-stripe** [comment]
This should be resolved now that we have the `stripe listen--print-secret` flag, which was implemented in #478 and released in [v1.4.5](https://github.com/stripe/stripe-cli/releases/tag/v1.4.5).
